File: src/main/java/com/yzj/alzassistant/controller/KnowledgeBaseController.java
package com.yzj.alzassistant.controller;

import com.mybatisflex.core.paginate.Page;
import com.yzj.alzassistant.annotation.AuthCheck;
import com.yzj.alzassistant.common.BaseResponse;
import com.yzj.alzassistant.common.ResultUtils;
import com.yzj.alzassistant.constant.UserConstant;
import com.yzj.alzassistant.exception.ThrowUtils;
import com.yzj.alzassistant.model.dto.knowledgeBase.KnowledgeBaseAddRequest;
import com.yzj.alzassistant.model.dto.knowledgeBase.KnowledgeBaseQueryRequest;
import com.yzj.alzassistant.model.dto.knowledgeBase.KnowledgeBaseUpdateRequest;
import com.yzj.alzassistant.model.entity.User;
import com.yzj.alzassistant.model.vo.KnowledgeBaseVO;
import com.yzj.alzassistant.service.KnowledgeBaseService;
import com.yzj.alzassistant.service.RagRebuildService;
import com.yzj.alzassistant.service.UserService;
import jakarta.annotation.Resource;
import jakarta.servlet.http.HttpServletRequest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

/**
 * 知识库文档 控制层。
 *
 * @author <a href="https://github.com/yozp">yunikon</a>
 */
@RestController
@RequestMapping("/knowledgeBase")
public class KnowledgeBaseController {

    @Resource
    private KnowledgeBaseService knowledgeBaseService;

    @Resource
    private UserService userService;

    @Resource
    private RagRebuildService ragRebuildService;

    /**
     * 文件上传并解析文档（单文件）
     */
    @PostMapping("/upload")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Long> uploadDocument(
            @RequestPart("file") MultipartFile file,
            @RequestParam(value = "category", required = false) String category,
            HttpServletRequest request) {
        ThrowUtils.throwIf(file == null || file.isEmpty(), com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR, "文件不能为空");
        User loginUser = userService.getLoginUser(request);
        Long id = knowledgeBaseService.addKnowledgeBaseByFile(file, category, loginUser);
        return ResultUtils.success(id);
    }

    /**
     * 多文件上传并解析文档
     */
    @PostMapping("/upload/multiple")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<List<Long>> uploadMultipleDocuments(
            @RequestPart("files") MultipartFile[] files,
            @RequestParam(value = "category", required = false) String category,
            HttpServletRequest request) {
        ThrowUtils.throwIf(files == null || files.length == 0, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR, "文件不能为空");
        User loginUser = userService.getLoginUser(request);
        List<Long> ids = knowledgeBaseService.addKnowledgeBaseByFiles(files, category, loginUser);
        return ResultUtils.success(ids);
    }

    /**
     * 添加知识库文档
     */
    @PostMapping("/add")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Long> addKnowledgeBase(@RequestBody KnowledgeBaseAddRequest addRequest, HttpServletRequest request) {
        ThrowUtils.throwIf(addRequest == null, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        Long id = knowledgeBaseService.addKnowledgeBase(addRequest, loginUser);
        return ResultUtils.success(id);
    }

    /**
     * 更新知识库文档
     */
    @PostMapping("/update")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Boolean> updateKnowledgeBase(@RequestBody KnowledgeBaseUpdateRequest updateRequest, HttpServletRequest request) {
        ThrowUtils.throwIf(updateRequest == null, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        boolean result = knowledgeBaseService.updateKnowledgeBase(updateRequest, loginUser);
        return ResultUtils.success(result);
    }

    /**
     * 删除知识库文档
     */
    @PostMapping("/delete")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Boolean> deleteKnowledgeBase(@RequestBody com.yzj.alzassistant.common.DeleteRequest deleteRequest, HttpServletRequest request) {
        ThrowUtils.throwIf(deleteRequest == null || deleteRequest.getId() == null, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR);
        User loginUser = userService.getLoginUser(request);
        boolean result = knowledgeBaseService.deleteKnowledgeBase(deleteRequest.getId(), loginUser);
        return ResultUtils.success(result);
    }

    /**
     * 分页查询知识库文档
     */
    @PostMapping("/list/page")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Page<KnowledgeBaseVO>> listKnowledgeBaseByPage(@RequestBody KnowledgeBaseQueryRequest queryRequest) {
        ThrowUtils.throwIf(queryRequest == null, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR);
        Page<KnowledgeBaseVO> page = knowledgeBaseService.listKnowledgeBaseByPage(queryRequest);
        return ResultUtils.success(page);
    }

    /**
     * 根据id获取知识库文档详情
     */
    @GetMapping("/get")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<KnowledgeBaseVO> getKnowledgeBaseById(@RequestParam Long id) {
        ThrowUtils.throwIf(id == null || id <= 0, com.yzj.alzassistant.exception.ErrorCode.PARAMS_ERROR);
        KnowledgeBaseVO vo = knowledgeBaseService.getKnowledgeBaseById(id);
        return ResultUtils.success(vo);
    }

    /**
     * 重新构建RAG向量库
     */
    @PostMapping("/rebuild-rag")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Boolean> rebuildRag() {
        boolean result = ragRebuildService.rebuildRag();
        return ResultUtils.success(result);
    }

}

File: alz-assistant-frontend/src/api/knowledgeBaseController.ts
// @ts-ignore
/* eslint-disable */
import request from '@/request'

/** 此处后端没有提供注释 POST /knowledgeBase/add */
export async function addKnowledgeBase(
  body: API.KnowledgeBaseAddRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseLong>('/knowledgeBase/add', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 POST /knowledgeBase/delete */
export async function deleteKnowledgeBase(
  body: API.DeleteRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseBoolean>('/knowledgeBase/delete', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 GET /knowledgeBase/get */
export async function getKnowledgeBaseById(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getKnowledgeBaseByIdParams,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseKnowledgeBaseVO>('/knowledgeBase/get', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 POST /knowledgeBase/list/page */
export async function listKnowledgeBaseByPage(
  body: API.KnowledgeBaseQueryRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponsePageKnowledgeBaseVO>('/knowledgeBase/list/page', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 POST /knowledgeBase/rebuild-rag */
export async function rebuildRag(options?: { [key: string]: any }) {
  return request<API.BaseResponseBoolean>('/knowledgeBase/rebuild-rag', {
    method: 'POST',
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 POST /knowledgeBase/update */
export async function updateKnowledgeBase(
  body: API.KnowledgeBaseUpdateRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseBoolean>('/knowledgeBase/update', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 POST /knowledgeBase/upload */
export async function uploadDocument(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.uploadDocumentParams,
  body: {},
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseLong>('/knowledgeBase/upload', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    params: {
      ...params,
    },
    data: body,
    ...(options || {}),
  })
}

/** 此处后端没有提供注释 POST /knowledgeBase/upload/multiple */
export async function uploadMultipleDocuments(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.uploadMultipleDocumentsParams,
  body: {},
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseListLong>('/knowledgeBase/upload/multiple', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    params: {
      ...params,
    },
    data: body,
    ...(options || {}),
  })
}
